eng/common-tests/README.md

```diff
@@ -0,0 +1,25 @@
+# About common-tests
+
+Every PowerShell source that is complex enough to have unit tests should have these tests be written in the [Pester](https://pester.dev/)
+framework and placed in `eng/common-tests/**`, i.e. this directory or one of its descendants.  
+By design, unlike `eng/common`, `eng/common-tests` is not subject to
+[mirroring to repositories](https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/common_engsys.md).
+
+## When tests in this directory are executed
+
+The PowerShell Pester unit tests will be executed by the [`tools - eng-common-tests`](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=220791) pipeline
```

Comment:
Does this need to be an internal pipeline? If it doesn't require secrets which I don't believe it does we should move it to public. That way we can have it automatically trigger for any PR.

Reply:
I created a public equivalent:
https://dev.azure.com/azure-sdk/public/_build?definitionId=5985&_a=summary

Once this PR is merged, I will delete the internal version:
https://dev.azure.com/azure-sdk/internal/_build?definitionId=5809

and also the internal `tools eng-script-tests`, which appears redundant to me:

https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=5823

Reply:
Yes we can combine them if they are doing similar things. 

For the public pipeline we generally add the "- ci" postfix. So `tools - eng-common-tests - ci` or `tools - script-tests - ci` if we make it more generic.

Reply:
Renamed to `tools - eng-common-tests - ci`. The `tools - script-tests` looks to be totally redundant to me, just a subset of `eng-common-tests`, so I will outright delete it.

Reply:
OK correction: the `script-tests` was running tests in `eng/scripts` but we have since moved all such tests to `eng/common-tests`

Reply:
OK I'm supportive of having just the one script testing pipeline.

Reply:
@weshaggard ungh, actually, there are tests, in:

`eng\scripts\contentreplacements\tests\Repo-File-Content-Replacements\Repo-File-Content-Replacements.Tests.ps1`

I was just running pester against `scripts` instead of `eng/scripts`. Should I move everything to `eng/scripts` ?

In any case, I cannot delete that pipeline yet. But I am thinking to take all the tests  in `eng/scripts` (which is just the one file above) and move them to `eng/common-tests` and rename `eng/common-tests` to e.g. `eng/powershell-tests`.

Does that sound reasonable?

If so, I'll make a separate PR for that, and for now hold off with any changes to the `tools - script-tests` pipeline.

Reply:
Two things.

### 1. PowerShell test pipeline setup

I had a chat with @weshaggard. We should keep multiple pipelines
- one pipeline for running PowerShell tests in `/eng/common-tests/`;
- one pipeline per each language repo for running PowerShell tests in its `/eng/scripts/` dir.

This is because:
- There are two separate sets of PowerShell scripts: `/eng/common/scripts/` and `/eng/scripts/` and both these sets need to have tests written for them.
- The dir `/eng/scripts/` in each language repo has language-repo-specific scripts; currently they hardly have any tests, but in the future they should have more tests.

The pipelines can be sourced from different `ci.yml` files - the `ci.yml` source would differ only by parameters passed to the underlying abstraction for running Pester tests. Most importantly, the `TargetDirectory` would point to appropriate dir.

### 2. PowerShell file naming conventions

Per chat in this PR and over Teams between me, @weshaggard and @benbp we want to adopt following convention for naming files:

- `/eng/scripts/foo.ps1` and `/eng/common/scripts/foo.ps1` for directly-invocable entry points;
- `/eng/scripts/foo.lib.ps1` and `/eng/common/scripts/foo.lib.ps1` for dot-sourceable functions; the `foo.ps1` is expected to be in practice just a delegate calling the "entry point" function from `foo.lib.ps1`;
- `/eng/scripts/foo.tests.ps1` and `/eng/common-tests/foo.tests.ps1` for Pester tests for `foo.lib.ps1`.

Reply:
Created an issue to track the work above:
- https://github.com/Azure/azure-sdk-tools/issues/5878